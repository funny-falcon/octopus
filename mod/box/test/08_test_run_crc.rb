#!/usr/bin/ruby1.9.1

$:.push 'test/lib'
require 'standalone_env'

class Env < StandAloneEnv
  def config
    super + <<EOD
run_crc_delay = 0.4

object_space[0].enabled = 1
object_space[0].index[0].type = "HASH"
object_space[0].index[0].unique = 1
object_space[0].index[0].key_field[0].fieldno = 0
object_space[0].index[0].key_field[0].type = "STR"
EOD
  end
end

def wait_for(n=100)
  n.times do
    return if yield
    sleep 0.05
  end
  raise "wait_for failed"
end

env = Env.clean
env.with_server do
  ping
  129.times do |i|
    insert [i, i + 1, i + 2]
  end

  wait_for { open('|./octopus --cat 00000000000000000002.xlog 2>/dev/null').lines.grep(/run_crc/).length > 0 }
  env.stop

  puts `./octopus --cat 00000000000000000002.xlog | sed 's/tm:[^ ]* //' | grep run_crc` + "\n"
end
