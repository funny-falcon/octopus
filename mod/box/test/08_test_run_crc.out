# box.ping()
:pong

# box.insert([0, 1, 2])
1

# box.insert([1, 2, 3])
1

# box.insert([2, 3, 4])
1

# box.insert([3, 4, 5])
1

# box.insert([4, 5, 6])
1

# box.insert([5, 6, 7])
1

# box.insert([6, 7, 8])
1

# box.insert([7, 8, 9])
1

# box.insert([8, 9, 10])
1

# box.insert([9, 10, 11])
1

# box.insert([10, 11, 12])
1

# box.insert([11, 12, 13])
1

# box.insert([12, 13, 14])
1

# box.insert([13, 14, 15])
1

# box.insert([14, 15, 16])
1

# box.insert([15, 16, 17])
1

# box.insert([16, 17, 18])
1

# box.insert([17, 18, 19])
1

# box.insert([18, 19, 20])
1

# box.insert([19, 20, 21])
1

# box.insert([20, 21, 22])
1

# box.insert([21, 22, 23])
1

# box.insert([22, 23, 24])
1

# box.insert([23, 24, 25])
1

# box.insert([24, 25, 26])
1

# box.insert([25, 26, 27])
1

# box.insert([26, 27, 28])
1

# box.insert([27, 28, 29])
1

# box.insert([28, 29, 30])
1

# box.insert([29, 30, 31])
1

# box.insert([30, 31, 32])
1

# box.insert([31, 32, 33])
1

# box.insert([32, 33, 34])
1

# box.insert([33, 34, 35])
1

# box.insert([34, 35, 36])
1

# box.insert([35, 36, 37])
1

# box.insert([36, 37, 38])
1

# box.insert([37, 38, 39])
1

# box.insert([38, 39, 40])
1

# box.insert([39, 40, 41])
1

# box.insert([40, 41, 42])
1

# box.insert([41, 42, 43])
1

# box.insert([42, 43, 44])
1

# box.insert([43, 44, 45])
1

# box.insert([44, 45, 46])
1

# box.insert([45, 46, 47])
1

# box.insert([46, 47, 48])
1

# box.insert([47, 48, 49])
1

# box.insert([48, 49, 50])
1

# box.insert([49, 50, 51])
1

# box.insert([50, 51, 52])
1

# box.insert([51, 52, 53])
1

# box.insert([52, 53, 54])
1

# box.insert([53, 54, 55])
1

# box.insert([54, 55, 56])
1

# box.insert([55, 56, 57])
1

# box.insert([56, 57, 58])
1

# box.insert([57, 58, 59])
1

# box.insert([58, 59, 60])
1

# box.insert([59, 60, 61])
1

# box.insert([60, 61, 62])
1

# box.insert([61, 62, 63])
1

# box.insert([62, 63, 64])
1

# box.insert([63, 64, 65])
1

# box.insert([64, 65, 66])
1

# box.insert([65, 66, 67])
1

# box.insert([66, 67, 68])
1

# box.insert([67, 68, 69])
1

# box.insert([68, 69, 70])
1

# box.insert([69, 70, 71])
1

# box.insert([70, 71, 72])
1

# box.insert([71, 72, 73])
1

# box.insert([72, 73, 74])
1

# box.insert([73, 74, 75])
1

# box.insert([74, 75, 76])
1

# box.insert([75, 76, 77])
1

# box.insert([76, 77, 78])
1

# box.insert([77, 78, 79])
1

# box.insert([78, 79, 80])
1

# box.insert([79, 80, 81])
1

# box.insert([80, 81, 82])
1

# box.insert([81, 82, 83])
1

# box.insert([82, 83, 84])
1

# box.insert([83, 84, 85])
1

# box.insert([84, 85, 86])
1

# box.insert([85, 86, 87])
1

# box.insert([86, 87, 88])
1

# box.insert([87, 88, 89])
1

# box.insert([88, 89, 90])
1

# box.insert([89, 90, 91])
1

# box.insert([90, 91, 92])
1

# box.insert([91, 92, 93])
1

# box.insert([92, 93, 94])
1

# box.insert([93, 94, 95])
1

# box.insert([94, 95, 96])
1

# box.insert([95, 96, 97])
1

# box.insert([96, 97, 98])
1

# box.insert([97, 98, 99])
1

# box.insert([98, 99, 100])
1

# box.insert([99, 100, 101])
1

# box.insert([100, 101, 102])
1

# box.insert([101, 102, 103])
1

# box.insert([102, 103, 104])
1

# box.insert([103, 104, 105])
1

# box.insert([104, 105, 106])
1

# box.insert([105, 106, 107])
1

# box.insert([106, 107, 108])
1

# box.insert([107, 108, 109])
1

# box.insert([108, 109, 110])
1

# box.insert([109, 110, 111])
1

# box.insert([110, 111, 112])
1

# box.insert([111, 112, 113])
1

# box.insert([112, 113, 114])
1

# box.insert([113, 114, 115])
1

# box.insert([114, 115, 116])
1

# box.insert([115, 116, 117])
1

# box.insert([116, 117, 118])
1

# box.insert([117, 118, 119])
1

# box.insert([118, 119, 120])
1

# box.insert([119, 120, 121])
1

# box.insert([120, 121, 122])
1

# box.insert([121, 122, 123])
1

# box.insert([122, 123, 124])
1

# box.insert([123, 124, 125])
1

# box.insert([124, 125, 126])
1

# box.insert([125, 126, 127])
1

# box.insert([126, 127, 128])
1

# box.insert([127, 128, 129])
1

# box.insert([128, 129, 130])
1

lsn:131 scn:131 t:2/run_crc 0.0.0.0:0 SCN:130 log:0xee3743f9

